drivers/MbedCRC.h

```diff
@@ -137,20 +148,18 @@ class MbedCRC {
     int32_t compute(void *buffer, crc_data_size_t size, uint32_t *crc)
     {
         MBED_ASSERT(crc != NULL);
-        int32_t status;
-        if (0 != (status = compute_partial_start(crc))) {
-            *crc = 0;
-            return status;
-        }
-        if (0 != (status = compute_partial(buffer, size, crc))) {
-            *crc = 0;
-            return status;
+        int32_t status = 0;
+        lock();
```

Comment:
What makes RAII reduce the code size?

Reply:
I was mostly referring to the line of code required but in the final binary; we can observe code size reduction too as the compiler is very good at optimizing destruction of objects on the stack.

Here you go: https://godbolt.org/g/nXarPA

Reply:
It looks like it will behave the same if you have a cleanup label to jump to: https://godbolt.org/z/rgagh-

But good to know gcc can do that automatically for RAII objects. It's a shame it can't figure out that the unlock() calls can be shared.